// react-intl-redux/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.update = exports.UPDATE = exports.Provider = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.intlReducer = intlReducer;

var _Provider2 = require('./components/Provider');

var _Provider3 = _interopRequireDefault(_Provider2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.Provider = _Provider3.default;
var UPDATE = exports.UPDATE = '@@intl/UPDATE';

var update = exports.update = function update(_ref) {
  var locale = _ref.locale;
  var messages = _ref.messages;

  return {
    type: UPDATE,
    payload: { locale: locale, messages: messages }
  };
};

var initialState = {
  locale: 'en',
  messages: {}
};

function intlReducer() {
  var state = arguments.length <= 0 || arguments[0] === undefined ? initialState : arguments[0];
  var action = arguments[1];

  if (action.type !== UPDATE) {
    return state;
  }

  return _extends({}, state, action.payload);
}
